### PR TITLE
Feature/viewr

### PR DESCRIPTION
**Pull request by AmeliaJo**
VIEWR input checking and tests

### CODE REVIEW

**Comment by AmeliaJo** [comment]
Closes #13

File: src/njoy21/input/VIEWR/Card1/Infile.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+struct Infile {
+  using Value_t = int;
+
+  static std::string name() { return "infile"; }
+
+  static std::string description() {
+	return
```
Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/Infile.hpp]: Please indent 2 space, not 4. See https://njoy.github.io/DevelopersGuide/Style/Code.html

File: src/njoy21/input/VIEWR/Card1/Infile/test/Infile.test.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#define CATCH_CONFIG_MAIN
+
+#include "catch.hpp"
+#include "njoy21.hpp"
+
+using namespace njoy::njoy21::input;
+
+SCENARIO( "infile input values" ){
+  GIVEN( "valid infile tape values" ){
+    std::vector<int> validValues{-20, 20, 42, 99, -99};
```
Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/Infile/test/Infile.test.cpp]: Nice job following the clean convention developed earlier.

File: src/njoy21/input/VIEWR/Card1/Nps.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  
+  using Value_t = int; 
+  
+  static std::string name(){ return "nps"; }
+  static std::string description(){
+    return
+      "The nps argument specifies the io unit used internally by the\n"
+      "Fortran routine and the filename to be output, e.g. setting nps to\n"
+      "21 or -21 specifies a file named 'tape21' is will be used for the\n"
+      "outputi.\n"
```
Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/Nps.hpp]: I think `outputi` should be `output`; minor typo.

File: src/njoy21/input/VIEWR/Card1/Nps.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      "ascii text files. Negative values indicate unformatted output i.e.\n"
+      "Fortran-style block binary.\n"
+      "\n"
+      "nps values are restricted to an absolute value between 20 and 99,\n"
+      "inclusively, and are required to have the same signedness as\n"
+      "(and different absolute value than) the corresponding nendf argument.";
+  }
+
+  static bool verify( const Value_t v, const Argument<Infile>& infile ){
+    return std::abs(v) >= 20 && std::abs(v) <= 99 && ( std::abs(infile.value) != std::abs( v ) );
```
Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/Nps.hpp]: Line lengths must be <= 80 characters. This one is too long.

File: src/njoy21/input/VIEWR/Card1/Nps/test/Nps.test.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          Argument< VIEWR::Card1::Infile > infile; 
+          infile.value = 22*(nps/std::abs(nps));
+          iRecordStream<char> issNps(
+              std::istringstream( std::to_string(nps) ) );
+          REQUIRE(nps == argument::extract< 
+                           VIEWR::Card1::Nps >(issNps, infile).value );
+        }
+      }
+    }
+    WHEN( "the infile value is the same as the nps value" ){
```
Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/Nps/test/Nps.test.cpp]: Nice job checking for this condition.

File: src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+SCENARIO( "Parsing valid VIEWR Card1 input" ){
+  { 
+    iRecordStream<char> iss( std::istringstream( "21 22" ) );
+    VIEWR::Card1 card1( iss );
+    REQUIRE( card1.infile.value == 21 );
+    REQUIRE( card1.nps.value == 22 );
+  }
+  {
+    iRecordStream<char> iss( std::istringstream( "21 -22") );
```
Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp]: I'm a little surprised that mixed mode (i.e., positive and negative values) can be used here; though, I can't seem to find anything in the documentation that disallows this.

**Comment by jlconlin** [comment]
Overall, a nice job. Just a few minor things to take care of.

File: src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp (hunk 6 above)
Comment (apmccartney) [inline src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp]: Documentation lies anyway. Do have a input deck handy with VIEWR to test with?

Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp]: I have used the following input deck in my NJOY class. 

```
plotr
 30                             / Card0
                                / Card1 
                                / Card2 
 'U-235 Total Cross Section'    / Card3 
 'From ENDF/B-VII.1'            / Card3a
                                / Card4 
                                / Card5 
                                / Card5a
                                / Card6 
                                / Card6a
 6 20 9228 3 1 0                / Card8 
                                / Card9
 99                             / Card2
viewr
 30 40
stop
```

Comment (apmccartney) [inline src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp]: This doesn't test mixed mode

Comment (apmccartney) [inline src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp]: @jlconlin

Comment (jlconlin) [inline src/njoy21/input/VIEWR/Card1/test/Card1.test.cpp]: I know it doesn't test mixed mode. It was just the input that was request.

**Comment by jlconlin** [comment]
Is it safe to delete this Pull Request as it has been superseded by the feature/DTFR Pull Request?

**Pull request merged**